ssl/ssl_lib.c

```diff
@@ -3396,7 +3393,7 @@ void SSL_CTX_free(SSL_CTX *a)
     SSL_CTX_SRP_CTX_free(a);
 #endif
 #ifndef OPENSSL_NO_ENGINE
```

Comment:
Is it worth folding this #ifndef into the tls_depr.c file too?

Reply:
Unfortunately the `client_cert_engine` field itself is only available if ENGINEs are available, so we still need the guard here.